graspologic/embed/case.py

```diff
@@ -0,0 +1,209 @@
+
+from graspologic.utils import import_graph, to_laplacian, is_almost_symmetric
+from graspologic.embed.base import BaseSpectralEmbed
+
+
+class CovariateAssistedEmbed(BaseSpectralEmbed):
+    """
+    Perform Spectral Embedding on a graph with covariates for each node, using the
+    regularized graph Laplacian.
+
+    The Covariate-Assisted Spectral Embedding is a k-dimensional Euclidean representation
+    of a graph. It returns an n x d matrix, similarly to Adjacency Spectral Embedding or
+    Laplacian Spectral Embedding. For more information, see [1].
+
+    Parameters
+    ----------
+    alpha : float, optional, default = None
+        Tuning parameter to use:
+            -  None: Default to the ratio of the leading eigenvector of the Laplacian
+                     to the leading eigenvector of the covariate matrix.
+            - float: Use a particular alpha-value.
+
+    assortative : bool, default = True
+        Embedding algorithm to use. An assortative network is any network where the
+        within-group probabilities are greater than the between-group probabilities.
+        Here, L is the regularized Laplacian, Y is the covariate matrix, and a is the
+        tuning parameter alpha:
+            - True: Embed ``L + a*Y@Y.T``. Better for assortative graphs.
+            - False: Embed ``L@L + a*Y@Y.T``. Better for non-assortative graphs.
+
+    n_components : int or None, default = None
+        Desired dimensionality of output data. If "full",
+        n_components must be <= min(Y.shape). Otherwise, n_components must be
+        < min(Y.shape). If None, then optimal dimensions will be chosen by
+        ``select_dimension`` using ``n_elbows`` argument.
+
+    n_elbows : int, optional, default: 2
+        If `n_components=None`, then compute the optimal embedding dimension using
+        `select_dimension`. Otherwise, ignored.
+
+    check_lcc : bool , optional, defult = True
+        Whether to check if input graph is connected. May result in non-optimal
+        results if the graph is unconnected. Not checking for connectedness may
+        result in faster computation.
+
+
+    References
+    ---------
+    .. [1] Binkiewicz, N., Vogelstein, J. T., & Rohe, K. (2017). Covariate-assisted
+    spectral clustering. Biometrika, 104(2), 361-377.
+    """
+
+    def __init__(
+        self,
+        alpha: Optional[float] = None,
+        assortative: bool = True,
+        n_components: Optional[int] = None,
+        n_elbows: int = 2,
+        check_lcc: bool = False,
+    ):
+        super().__init__(
+            n_components=n_components,
+            n_elbows=n_elbows,
+            check_lcc=check_lcc,
+            concat=False,
+            algorithm="eigsh",
+        )
+
+        if not isinstance(assortative, bool):
+            msg = "embedding_alg must be a boolean value."
+            raise ValueError(msg)
+        self.assortative = assortative
+
+        if not ((alpha is None) or isinstance(alpha, (float, int))):
```

Comment:
```suggestion
        if not isinstance(alpha, (float, int, type(None))):
```
You could do this if you want to make the line slightly shorter

Reply:
thanks, 8c8d974